xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/displayers/defaultDisplayer.js

```diff
@@ -0,0 +1,221 @@
+
+    this.element = undefined;
+    this.initElement();
+
+  };
+
+
+  /**
+   * Create the root element of the displayer
+   */
+  Displayer.prototype.initElement = function () {
+    var self = this;
+    if (this.element) { return; }
+
+    this.element = document.createElement("div");
+    // set attributes
+    this.element.className = "livedata-displayer";
+    this.element.tabIndex = "0";
+    // bind events (switching to edit mode)
+    this.element.ondblclick = function () {
+      if (self.isView) {
+        self.edit();
+      }
+    };
+    this.element.onkeydown = function (e) {
+      if (e.keyCode === 13 /*ENTER*/ && e.target === self.element && self.isView) {
+        self.edit();
+      }
+    };
+  };
+
+
+  /**
+   * Return the object of parameters to be passed to the viewer and editor functions
+   * @returns {Object} an object containing useful data for displayer: {
+   *  value: the entry property value to be displayed
+   *  property: the property descriptor object
+   *  entry: the entry data object
+   *  config: the configuration object of the displayer, found in the propertyDescriptor
+   *  data: the livedata data object
+   *  logic: the logic instance
+   * }
+   */
+  Displayer.prototype._createParameters = function () {
+    // return the param object
+    return {
+      value: this.entryData[this.propertyId],
+      property: this.logic.getPropertyDescriptor(this.propertyId),
+      entry: this.entryData,
+      config: this.logic.getDisplayerDescriptor(this.propertyId),
+      data: this.logic.data,
+      logic: this.logic,
+    };
+  };
+
+
+
+  /**
+   * Create viewer element for the displayer
+   * This method can be overriden by other displayers that inherit from this one
+   * Parameters are given by the Displayer.prototype.view function
+   * @param {object} params An object containing useful data for the displayer.
+   *  Param object detail can be found in the Displayer.prototype._createParameters method
+   * @returns {Promise}
+   */
+  Displayer.prototype.createView = function (params) {
+    return new Promise (function (resolve, reject) {
+      var element = document.createElement("div");
+      if (params.value !== undefined && params.value !== null) {
+        element.innerText = params.value;
+      }
+      resolve(element);
+    });
+  };
+
+
+
+  /**
+   * Create editor element for the displayer
+   * This method can be overriden by other displayers that inherit from this one
+   * Parameters are given by the Displayer.prototype.edit function
+   * @param {object} params An object containing useful data for the displayer.
+   *  Param object detail can be found in the Displayer.prototype._createParameters method
+   * @returns {Promise}
+   */
+  Displayer.prototype.createEdit = function (params) {
+    var self = this;
+    return new Promise (function (resolve, reject) {
+      var input = document.createElement("input");
+      input.size = 1;
+      input.style.width = "100%";
```

Comment:
Can't this be put in the style sheet?

Reply:
I can put this in the `displayer.css` stylesheet, as this is the default displayer css. But are we going to put every displayer custom style in the same stylesheet? If we are separating the displayers in different files, we might want to do the same with their css.
If the css is in a global stylesheet we will need to prevent a displayer custom css modifying other ones, by adding the displayer id in the class name of its element.

Reply:
> But are we going to put every displayer custom style in the same stylesheet?

We can, but I think it's overkill. I would put the styles for the standard displayers, that we provide by default, in a single stylesheet because I don't think they need that much custom styles for now.

> If the css is in a global stylesheet we will need to prevent a displayer custom css modifying other ones, by adding the displayer id in the class name of its element.

You need this no matter how you split the CSS in files, because different properties can use different displayers so we definitely need to load the CSS for multiple displayers and apply it only where it make sense.